integration/cluster_test.go

```diff
@@ -416,3 +419,61 @@ func clusterMustProgress(t *testing.T, membs []*member) {
 		mcancel()
 	}
 }
+
+func TestStopTriggerTransferLeader(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	clus := NewClusterV3(t, &ClusterConfig{Size: 3, AutoLeaderTransfer: true})
+	defer clus.Terminate(t)
+
+	leaderIdx1 := clus.waitLeader(t, clus.Members)
+	leaderID1 := clus.Members[leaderIdx1].s.ID()
+	followerIdx := (leaderIdx1 + 1) % 3
+
+	donec, stopc := make(chan struct{}), make(chan struct{})
+	go func() {
+		defer close(donec)
+
+		kvc := toGRPC(clus.Client(followerIdx)).KV
+		for {
+			select {
+			case <-stopc:
+				return
+			default:
+				// 10-second is enough timeout to cover leader failure
+				sctx, scancel := context.WithTimeout(context.TODO(), 10*time.Second)
+				_, err := kvc.Range(sctx, &etcdserverpb.RangeRequest{Key: []byte("foo")})
+				scancel()
+				if err != nil {
+					switch grpc.ErrorDesc(err) {
+					case etcdserver.ErrTimeoutDueToLeaderFail.Error(), etcdserver.ErrTimeout.Error():
```

Comment:
If this error is happening on a controlled transfer, then it's still wrong. I think this would get the same error even without transfer support and the test would pass.